javadoc build issues

I've tryied to build jar with dependencies just like that:

```
git clone git@github.com:yandex/clickhouse-jdbc.git
cd clickhouse-jdbc
docker run -it --rm -v "$(pwd)":/usr/src/mymaven -w /usr/src/mymaven maven mvn package assembly:single -DskipTests=true
```

It fails during javadoc build:
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:59 min
[INFO] Finished at: 2019-07-12T07:09:28Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.9.1:jar (attach-javadocs) on project clickhouse-jdbc: MavenReportException: Error while creating archive: 
[ERROR] Exit code: 1 - javadoc: error - The code being documented uses modules but the packages defined in http://docs.oracle.com/javase/6/docs/api/ are in the unnamed module.
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/ClickHouseDriver.java:116: warning: no description for @param
[ERROR]      * @param rate
[ERROR]        ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/ClickHouseDriver.java:117: warning: no description for @param
[ERROR]      * @param timeUnit
[ERROR]        ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/ClickHouseDataSource.java:113: warning: no description for @param
[ERROR]      * @param rate
[ERROR]        ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/ClickHouseDataSource.java:114: warning: no description for @param
[ERROR]      * @param timeUnit
[ERROR]        ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/response/FastByteArrayOutputStream.java:120: error: tag not supported in the generated HTML version: tt
[ERROR]      * Closing a <tt>ByteArrayOutputStream</tt> has no effect. The methods in
[ERROR]                  ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/response/FastByteArrayOutputStream.java:122: error: tag not supported in the generated HTML version: tt
[ERROR]      * generating an <tt>IOException</tt>.
[ERROR]                      ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/response/FastByteArrayOutputStream.java:123: warning: empty <p> tag
[ERROR]      * <p>
[ERROR]        ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/response/FastByteArrayOutputStream.java:159: warning: no @return
[ERROR]     public FastByteArrayInputStream convertToInputStream() {
[ERROR]                                     ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseArrayUtil.java:22: warning: no @return
[ERROR]     public static String arrayToString(Object object, Boolean explicitEscape) {
[ERROR]                          ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseArrayUtil.java:147: warning: no @return
[ERROR]     public static String toString(Collection collection, boolean escape) {
[ERROR]                          ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryInputStream.java:106: warning: no description for @throws
[ERROR] 	 * @throws IOException
[ERROR] 	   ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryInputStream.java:123: warning: no description for @throws
[ERROR] 	 * @throws IOException
[ERROR] 	   ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryInputStream.java:140: warning: no description for @throws
[ERROR] 	 * @throws IOException
[ERROR] 	   ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryInputStream.java:153: warning: no description for @throws
[ERROR] 	 * @throws IOException
[ERROR] 	   ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryStream.java:54: warning: no description for @param
[ERROR]      * @param bytes
[ERROR]        ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryStream.java:55: warning: no description for @throws
[ERROR]      * @throws IOException
[ERROR]        ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryStream.java:64: warning: no description for @param
[ERROR]      * @param b
[ERROR]        ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryStream.java:65: warning: no description for @throws
[ERROR]      * @throws IOException
[ERROR]        ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryStream.java:292: warning: no @param for isNullable
[ERROR]     public void markNextNullable(boolean isNullable) throws IOException {
[ERROR]                 ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryStream.java:292: warning: no @throws for java.io.IOException
[ERROR]     public void markNextNullable(boolean isNullable) throws IOException {
[ERROR]                 ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/Utils.java:27: warning: no @param for str
[ERROR]     public static String[] splitWithoutEscaped(String str, char separatorChar, boolean retainEmpty) {
[ERROR]                            ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/Utils.java:27: warning: no @param for separatorChar
[ERROR]     public static String[] splitWithoutEscaped(String str, char separatorChar, boolean retainEmpty) {
[ERROR]                            ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/Utils.java:27: warning: no @param for retainEmpty
[ERROR]     public static String[] splitWithoutEscaped(String str, char separatorChar, boolean retainEmpty) {
[ERROR]                            ^
[ERROR] /usr/src/mymaven/src/main/java/ru/yandex/clickhouse/util/Utils.java:27: warning: no @return
[ERROR]     public static String[] splitWithoutEscaped(String str, char separatorChar, boolean retainEmpty) {
[ERROR]                            ^
[ERROR] 
[ERROR] Command line was: /usr/local/openjdk-11/bin/javadoc @options @packages
[ERROR] 
[ERROR] Refer to the generated Javadoc files in '/usr/src/mymaven/target/apidocs' dir.
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

Workaround is trivial - i've added `-Dmaven.javadoc.skip=true` and everything's ok. But it's better to make build option listed in README working.

--- a/src/main/java/ru/yandex/clickhouse/ClickHouseDataSource.java
+++ b/src/main/java/ru/yandex/clickhouse/ClickHouseDataSource.java
@@ -110,9 +110,9 @@ public class ClickHouseDataSource implements DataSource {
      * Schedules connections cleaning at a rate. Turned off by default.
      * See https://hc.apache.org/httpcomponents-client-4.5.x/tutorial/html/connmgmt.html#d5e418
      *
-     * @param rate
-     * @param timeUnit
-     * @return this
+     * @param rate period when checking would be performed
+     * @param timeUnit time unit of rate
+     * @return current modified object
      */
     public ClickHouseDataSource withConnectionsCleaning(int rate, TimeUnit timeUnit){
         driver.scheduleConnectionsCleaning(rate, timeUnit);

--- a/src/main/java/ru/yandex/clickhouse/ClickHouseDriver.java
+++ b/src/main/java/ru/yandex/clickhouse/ClickHouseDriver.java
@@ -113,8 +113,9 @@ public class ClickHouseDriver implements Driver {
     /**
      * Schedules connections cleaning at a rate. Turned off by default.
      * See https://hc.apache.org/httpcomponents-client-4.5.x/tutorial/html/connmgmt.html#d5e418
-     * @param rate
-     * @param timeUnit
+     *
+     * @param rate period when checking would be performed
+     * @param timeUnit time unit of rate
      */
     public void scheduleConnectionsCleaning(int rate, TimeUnit timeUnit){
         ScheduledConnectionCleaner.INSTANCE.scheduleAtFixedRate(new Runnable() {

--- a/src/main/java/ru/yandex/clickhouse/response/FastByteArrayOutputStream.java
+++ b/src/main/java/ru/yandex/clickhouse/response/FastByteArrayOutputStream.java
@@ -120,8 +120,6 @@ public final class FastByteArrayOutputStream extends OutputStream {
      * Closing a <tt>ByteArrayOutputStream</tt> has no effect. The methods in
      * this class can be called after the stream has been closed without
      * generating an <tt>IOException</tt>.
-     * <p>
-     *
      */
     @Override
     public void close() throws IOException {
@@ -155,6 +153,7 @@ public final class FastByteArrayOutputStream extends OutputStream {
 
     /**
      * Creates InputStream using the same data that is written into this stream with no copying in memory
+     * @return a input stream contained all bytes recorded in a current stream
      */
     public FastByteArrayInputStream convertToInputStream() {
         return new FastByteArrayInputStream(buf, count); 

--- a/src/main/java/ru/yandex/clickhouse/util/ClickHouseArrayUtil.java
+++ b/src/main/java/ru/yandex/clickhouse/util/ClickHouseArrayUtil.java
@@ -18,6 +18,7 @@ public class ClickHouseArrayUtil {
     /**
      * @param object         the object to convert to ClickHouse-string representation
      * @param explicitEscape enable or disable elements escaping (works only for non-primitive values)
+     * @return string representation of an object
      */
     public static String arrayToString(Object object, Boolean explicitEscape) {
         if (!object.getClass().isArray()) {
@@ -143,6 +144,7 @@ public class ClickHouseArrayUtil {
      *
      * @param collection the collection to transform
      * @param escape     enable or disable escaping of the collection elements
+     * @return string representation of a collection
      */
     public static String toString(Collection collection, boolean escape) {
         return toString(collection.toArray(), escape);

--- a/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryInputStream.java
+++ b/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryInputStream.java
@@ -102,8 +102,9 @@ public class ClickHouseRowBinaryInputStream implements Closeable {
 
 	/**
 	 * Warning: the result is negative in Java if UInt8 &gt; 0x7f
+     *
 	 * @return next UInt8 value as a byte
-	 * @throws IOException
+	 * @throws IOException in case if an I/O error occurs
 	 */
 	public byte readUInt8AsByte() throws IOException {
 		return in.readByte();
@@ -120,7 +121,7 @@ public class ClickHouseRowBinaryInputStream implements Closeable {
 	/**
 	 * Warning: the result is negative in Java if UInt16 &gt; 0x7fff
 	 * @return next UInt16 value as a short
-	 * @throws IOException
+	 * @throws IOException in case if an I/O error occurs
 	 */
 	public short readUInt16AsShort() throws IOException {
 		return in.readShort();
@@ -137,7 +138,7 @@ public class ClickHouseRowBinaryInputStream implements Closeable {
 	/**
 	 * Warning: the result is negative in Java if UInt32 &gt; 0x7fffffff
 	 * @return next UInt32 value as an int
-	 * @throws IOException
+	 * @throws IOException in case if an I/O error occurs
 	 */
 	public int readUInt32AsInt() throws IOException {
 		return in.readInt();
@@ -150,7 +151,7 @@ public class ClickHouseRowBinaryInputStream implements Closeable {
 	/**
 	 * Warning: the result is negative in Java if UInt64 &gt; 0x7fffffffffffffff
 	 * @return next UInt64 value as a long
-	 * @throws IOException
+	 * @throws IOException in case if an I/O error occurs
 	 */
 	public long readUInt64AsLong() throws IOException {
 		return in.readLong();

--- a/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryStream.java
+++ b/src/main/java/ru/yandex/clickhouse/util/ClickHouseRowBinaryStream.java
@@ -51,8 +51,8 @@ public class ClickHouseRowBinaryStream {
     /**
      * Dangerous. Can only be used for rare optimizations, for example when the string is written in parts
      * without prior concatenation. The size of the string in bytes must be passed through writeUnsignedLeb128.
-     * @param bytes
-     * @throws IOException
+     * @param bytes byte array will be written into stream
+     * @throws IOException in case if an I/O error occurs
      */
     public void writeBytes(byte[] bytes) throws IOException {
         out.write(bytes);
@@ -61,8 +61,9 @@ public class ClickHouseRowBinaryStream {
     /**
      * Dangerous. Can only be used for rare optimizations, for example when the string is written in parts
      * without prior concatenation. The size of the string in bytes must be passed through writeUnsignedLeb128.
-     * @param b
-     * @throws IOException
+     *
+     * @param b byte value
+     * @throws IOException in case if an I/O error occurs
      */
     public void writeByte(byte b) throws IOException {
         out.write(b);
@@ -278,7 +279,7 @@ public class ClickHouseRowBinaryStream {
      *
      * E.g., to write Nullable(Int32):
      *
-     * {{{
+     * <pre>
      *     void writeNullableInt32(Integer value) {
      *         if (value == null) {
      *             markNextNullable(true);
@@ -287,7 +288,10 @@ public class ClickHouseRowBinaryStream {
      *             writeInt32(value);
      *         }
      *     }
-     * }}}
+     * </pre>
+     *
+     * @param isNullable if it's true, 1 will be written otherwise 0
+     * @throws IOException in case if an I/O error occurs
      */
     public void markNextNullable(boolean isNullable) throws IOException {
         writeByte(isNullable ? (byte) 1 : (byte) 0);

--- a/src/main/java/ru/yandex/clickhouse/util/Utils.java
+++ b/src/main/java/ru/yandex/clickhouse/util/Utils.java
@@ -22,7 +22,12 @@ public class Utils {
     }
 
     /**
-     * does not take into account escaped separators
+     * Does not take into account escaped separators
+     *
+     * @param str  the String to parse, may be null
+     * @param separatorChar  the character used as the delimiter
+     * @param retainEmpty if it is true, result can contain empty strings
+     * @return string array
      */
     public static String[] splitWithoutEscaped(String str, char separatorChar, boolean retainEmpty) {
         int len = str.length();

